src/app/model/car-impl.ts

```diff
@@ -0,0 +1,60 @@
+import { Vehicle } from "./vehicle";
+import { World } from "./world";
+import { Strategy } from "./strategy";
+
+export class Car implements Vehicle {
+  x: number;
+  y: number;
+  targetX: number;
+  targetY: number;
+  size: number;
+  ref;
+  strategy: Strategy;
+
+  constructor(
+    x: number,
+    y: number,
+    targetX: number,
+    targetY: number,
+    size: number,
+    ref,
+    strategy: Strategy,
+  ) {
+    this.x = x;
+    this.y = y;
+    this.targetX = targetX;
+    this.targetY = targetY;
+    this.size = size;
+    this.ref = ref;
+    this.strategy = strategy;
+    this.strategy.initialize(this);
+  }
+
+  getX(): number {
+    return this.x;
+  }
+
+  getY(): number {
+    return this.y;
+  }
+
+  getTargetX(): number {
+    return this.targetX;
+  }
+
+  getTargetY(): number {
+    return this.targetY;
+  }
+
+  getSize(): number {
+    return this.size;
+  }
```

Comment:
Maybe consider variable accesses instead of getters? Especially if all behavior for the car is provided in another class through the behvaior